### ISSUE TITLE
OmniAuth/Other LMS tools

### ISSUE THREAD

**Issue by Dannymh** [mit-teaching-systems-lab/discourse-edx-lti#31]
Hi,

I am looking to adapt what you have created here and build a more generic LTI tool to make it available from other LMS tools. I can get the launch to work, but get the "oops this page does not exist" errors.

What I am trying to figure out from the code is how the omniauth is triggered and on what side? Is that triggered from edX or from discourse and would discourse therefore need omniauth plugin installed as a requirement?

Dan

**Comment by kevinrobinson** [comment]
@Dannymh Yep, this is a Discourse plugin that handles the LTI launch.  The way this setup works is shown in the images in the [README](https://github.com/mit-teaching-systems-lab/discourse-edx-lti), that might help to see how the pieces fit together.  Essentially the learner starts in EdX, EdX does an "LTI launch" over to the Discourse app.  And this repo is a Discourse plugin that teaches Discourse how to receive that LTI launch and log in the user.

Am I following, you're trying to have another LMS tool do an LTI launch into Discourse?  If that's the case, I'd guess that this would mostly just work but haven't tried myself.  If you want to share more information about what particular error messages you're getting from this plugin, or what LMS you're trying to connect to and how it works I can try to help troubleshoot.  But I haven't done this and so might not have the best advice for you :)
